### PR TITLE
helm: added `PodDisruptionBudget` named template

### LINKED ISSUE #2292: Helm: Use a named template to define `PodDisruptionBudget` resources
#### Is your feature request related to a problem? Please describe.

Most of the `PodDisruptionBudget` definitions look the same across all different template components. This is very error prone and makes helm charts hard to maintain. 

#### Describe the solution you'd like

Define a new named template that generates the resource manifest and replace all definitions with this new method. 

#### Describe alternatives you've considered

Leaving it as it is now, with all the drawbacks this may entail. (verbosity, bad maintainability, etc.) 

#### Additional context

N/A

### PR DESCRIPTION
**Pull request by ortuman**
Signed-off-by: Miguel Ángel Ortuño <ortuman@gmail.com>

#### What this PR does

This PR makes use of a named template to render `PodDisruptionBudget` manifests. 

Interestingly, after running `make build-helm-tests` locally a bug has been brougth to light. Turns out that the `store-gateway` PDB resource was being rendered **always**, regardless of the actual `store-gateway.podDisruptionBudget` value. Reason was that the template was [referencing](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml#L1) `store_gateway` key instead of the actual `store-gateway`.  

This type of error is one of the reasons that motivates this change.

#### Which issue(s) this PR fixes or relates to

Fixes #2292 

#### Checklist

- [X] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/templates/nginx/nginx-pdb.yaml [hunk 1]
```diff
@@ -1,16 +1,3 @@
...
-  labels:
-    {{- include "mimir.labels" (dict "ctx" . "component" "nginx") | nindent 4 }}
-  namespace: {{ .Release.Namespace | quote }}
-spec:
-  selector:
-    matchLabels:
-      {{- include "mimir.selectorLabels" (dict "ctx" . "component" "nginx") | nindent 6 }}
-{{ toYaml .Values.nginx.podDisruptionBudget | indent 2 }}
-{{- end -}}
+{{- include "mimir.lib.podDisruptionBudget" (dict "ctx" $ "component" "nginx") }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/nginx/nginx-pdb.yaml:2]: Should we also check `nginx.enabled`?

File: operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{{- define "mimir.lib.podDisruptionBudget" -}}
+{{- $componentSection := include "mimir.componentSectionFromName" . }}
+{{ with (index $.ctx.Values $componentSection) }}
+{{- if .podDisruptionBudget -}}
+apiVersion: {{ include "mimir.podDisruptionBudget.apiVersion" $.ctx }}
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" $.ctx "component" $.component) }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" $.ctx "component" $.component "memberlist" $.memberlist) | nindent 4 }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl:17]: Should be be adding `app.kubernetes.io/part-of: memberlist` to a PodDisruptionBudget? IMO no. WDYT?

File: operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  namespace: {{ $.ctx.Release.Namespace | quote }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "mimir.selectorLabels" (dict "ctx" $.ctx "component" $.component "memberlist" $.memberlist) | nindent 6 }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl:22]: `memberlist` here doesn't do anything, I think we should remove it

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl:22]:
```suggestion
      {{- include "mimir.selectorLabels" (dict "ctx" $.ctx "component" $.component) | nindent 6 }}
```

**Review by dimitarvdimitrov — COMMENTED** [review]
looks good, thanks for adopting this PR, left two questions

File: operations/helm/charts/mimir-distributed/templates/nginx/nginx-pdb.yaml (hunk 1 above)
Comment (krajorama) [reply operations/helm/charts/mimir-distributed/templates/nginx/nginx-pdb.yaml:2]: nice catch

File: operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl (hunk 3 above)
Comment (krajorama) [reply operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl:22]: done

File: operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl (hunk 2 above)
Comment (krajorama) [reply operations/helm/charts/mimir-distributed/templates/lib/pdb.tpl:17]: yes, not much point. I think we're adding that label in some other places as well uselessly. Doesn't actually hurt as far as I know.

**Comment by krajorama** [comment]
trivial rebase

**Review by dimitarvdimitrov — APPROVED** [review]

**Pull request merged**